extension/src/install.ts

```diff
@@ -89,18 +92,36 @@ export async function updateExtensions(extensionInfo: ExtensionInfoService, pack
         await showReloadPrompt(
             localize(
                 'reload.to.complete.update.all',
-                'Please reload Visual Studio Code to complete updating the extensions.',
+                'to complete updating the extensions.',
             ),
         );
     }
 }
 
 /**
  * Displays a message with a button to reload vscode.
- * @param message The message to display.
+ * @param suffixMessage The reason to reload VSCode. The message will be displayed in a "popin".
  */
-export async function showReloadPrompt(message: string): Promise<void> {
-    const reload = await vscode.window.showInformationMessage(message, localize('reload.now', 'Reload Now'));
+export async function showReloadPrompt(suffixMessage: string): Promise<void> {
+    let reload: any;
+
+    if (getAutoReload()) {
+        const message = localize(
+            'automatic.reload',
+            'Visual Studio Code will restart in 3 secondes {0}.',
+            suffixMessage,
+        );
+        vscode.window.showInformationMessage(message);
```

Comment:
VS Code is usually pretty good about preserving state across a reload, but I would still be concerned about potential data loss. What if we lengthened the timer and provided an option to cancel the reload?

I think something like this might work:
```ts
const CANCEL_MESSAGE = localize('cancel', 'Cancel');
// ...

    const cancel = vscode.window.showInformationMessage(message, CANCEL_MESSAGE);
    const timeout = sleep(10000);
    const result = await Promise.race([cancel, timeout]);
    reload = result === CANCEL_MESSAGE;
```

Reply:
I run some test locally, when user doesn't have the autoSave, the change are not lost.
But I like you idea to provide a cancel option :)